src/Raven.Server/Documents/Indexes/Static/CurrentIndexingScope.cs

```diff
@@ -222,7 +232,21 @@ public unsafe dynamic LoadDocument(LazyStringValue keyLazy, string keyString, st
                 if (document.TryGetMetadata(out var metadata) && metadata.TryGet(Constants.Documents.Metadata.Collection, out string collection))
                 {
                     if (string.Equals(collection, collectionName, StringComparison.OrdinalIgnoreCase) == false)
+                    {
+                        if (_mismatchedReferencesHandler.MismatchedReferences.Count < MismatchedReferencesHandler.MaxMismatchedDocumentLoadsPerIndex)
+                        {
+                            _mismatchedReferencesHandler._lastLoadMismatched = true;
+                            _mismatchedReferencesHandler.HandleMismatchedReference(document, collectionName, id, collection);
```

Comment:
Shoudn't `_lastLoadMismatched = true;` be set by `HandleMismatchedReference()`?

Reply:
Done